lib/auth/auth_with_roles.go

```diff
@@ -3420,6 +3454,16 @@ func (a *ServerWithRoles) UpsertUser(ctx context.Context, u types.User) (types.U
 		return nil, trace.Wrap(err)
 	}
 
+	if a.hasBuiltinRole(types.RoleOkta) {
```

Comment:
Considering this conditional is repeated it might be worth putting into a function